jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -381,14 +386,26 @@ def _interaction_in_ext_step(self, event={}):
             self.marks['trace'].update_xy(xs,
                                           trace.trace)
             self.marks['trace'].line_style = 'dashed'
-            self.marks['ext_lower'].update_xy(xs,
-                                              trace.trace-self.ext_width/2)
-            self.marks['ext_upper'].update_xy(xs,
-                                              trace.trace+self.ext_width/2)
+            if self.ext_type_selected == 'Boxcar':
+                self.marks['ext_lower'].update_xy(xs,
+                                                  trace.trace-self.ext_width/2)
+                self.marks['ext_upper'].update_xy(xs,
+                                                  trace.trace+self.ext_width/2)
+            else:
+                for mark in ['ext_lower', 'ext_upper']:
+                    self.marks[mark].clear()
 
         self.active_step = 'ext'
         self._update_plugin_marks()
 
+        # TODO: remove this, the traitlet, and the row in spectral_extraction.vue
+        # when specutils handles the warning/exception
+        if self.ext_type_selected == 'Horne':
+            inp_sp2d = self._get_ext_input_spectrum()
+            self.ext_uncert_warn = inp_sp2d.uncertainty is not None and not hasattr(inp_sp2d.uncertainty, 'uncertainty_type')  # noqa
```

Comment:
To cover the numpy array uncertainty case we've been discussing, I believe you need something like `and inp_sp2d.uncertainty != 'unknown'` at the end of this statement. I tested by setting `inp_sp2d.uncertainty = np.random.rand(*inp_sp2d.flux.shape)` just above it, and the warning doesn't display in Specviz2D with this check as it's currently constructed.

Reply:
ah, it seems specutils is wrapping the bare array in an `UnknownUncertainty` object.  I changed the logic to just check for that explicitly now.

Reply:
That's a better way to go about it, and it works for me now.